src/smefit/runner.py

```diff
@@ -104,21 +104,57 @@ def ns(self):
         comm = MPI.COMM_WORLD
         rank = comm.Get_rank()
 
-        if rank == 0:
-            config = self.run_card
-            opt = NSOptimizer.from_dict(config)
+        config = self.run_card
+
+        # single parameter fits
+        if config["single_parameter_fits"]:
+            for coeff in config["coefficients"].keys():
+                single_coeff_config = dict(config)
+                single_coeff_config["coefficients"] = {}
+                single_coeff_config["coefficients"][coeff] = config["coefficients"][
+                    coeff
+                ]
+                if rank == 0:
+                    opt = NSOptimizer.from_dict(single_coeff_config)
+                else:
+                    opt = None
+
+                # Run optimizer
+                opt = comm.bcast(opt, root=0)
+                opt.run_sampling()
+
+        # global fit
         else:
-            opt = None
 
-        # Run optimizer
-        opt = comm.bcast(opt, root=0)
-        opt.run_sampling()
+            if rank == 0:
+                opt = NSOptimizer.from_dict(config)
+            else:
+                opt = None
+
+            # Run optimizer
+            opt = comm.bcast(opt, root=0)
+            opt.run_sampling()
 
     def mc(self):
         """
         Run a fit with MC
         """
         config = self.run_card
-        opt = MCOptimizer.from_dict(config)
-        opt.run_sampling()
-        opt.save()
+
+        # single parameter fits
+        if config["single_parameter_fits"]:
```

Comment:
also here maybe you can avoid repetition...

Reply:
mm you r right maybe then I should code another function just for the single parameter fits case